pallets/loans-ref/src/types.rs

```diff
@@ -635,6 +689,41 @@ impl<T: Config> ActiveLoan<T> {
 		Ok(())
 	}
 
+	fn ensure_can_modify(&self, mutation: &Mutation<T::Rate>) -> DispatchResult {
+		// What happens if it has overdue? or has been repaid or written off?
+
+		todo!()
+	}
+
+	pub fn modify_with(&mut self, mutation: Mutation<T::Rate>) -> DispatchResult {
+		self.ensure_can_modify(&mutation)?;
+
+		// TODO: With #1207 it will create ActiveLoanMutations instead of modifying the loan.
+
+		match mutation {
+			Mutation::Maturity(maturity) => self.info.schedule.maturity = maturity,
+			Mutation::InterestRate(rate) => self.update_interest_rate(rate)?,
+			Mutation::InterestPayments(payments) => self.info.schedule.interest_payments = payments,
+			Mutation::PayDownSchedule(schedule) => self.info.schedule.pay_down_schedule = schedule,
+			Mutation::ValuationMethod(method) => self.info.valuation_method = method,
+			Mutation::ProbabilityOfDefault(rate) => {
+				self.info
+					.mut_validation_method_dcf::<T>()?
+					.probability_of_default = rate
+			}
+			Mutation::LossGivenDefault(rate) => {
+				self.info
+					.mut_validation_method_dcf::<T>()?
+					.loss_given_default = rate
+			}
+			Mutation::DiscountRate(rate) => {
+				self.info.mut_validation_method_dcf::<T>()?.discount_rate = rate
+			}
```

Comment:
We should definietly update the valuation here

Reply:
It's already done because `modify()` is called within the `update_active_loan()`